rllib/evaluation/postprocessing.py

```diff
@@ -16,6 +16,55 @@ class Postprocessing:
     VALUE_TARGETS = "value_targets"
 
 
+def adjust_nstep(n_step: int, gamma: float, batch: SampleBatch) -> None:
+    """Rewrites `batch` to encode n-step rewards, dones, and next-obs.
+
+    Observations and actions remain unaffected. At the end of the trajectory,
+    n is truncated to fit in the traj length.
+
+    Args:
+        n_step (int): The number of steps to look ahead and adjust.
+        gamma (float): The discount factor.
+        batch (SampleBatch): The SampleBatch to adjust (in place).
+
+    Examples:
+        n-step=3
+        Trajectory=o0 r0 d0, o1 r1 d1, o2 r2 d2, o3 r3 d3, o4 r4 d4=True o5
+        gamma=0.9
+        Returned trajectory:
+        0: o0 [r0 + 0.9*r1 + 0.9^2*r2 + 0.9^3*r3] d3 o0'=o3
+        1: o1 [r1 + 0.9*r2 + 0.9^2*r3 + 0.9^3*r4] d4 o1'=o4
+        2: o2 [r2 + 0.9*r3 + 0.9^2*r4] d4 o1'=o5
+        3: o3 [r3 + 0.9*r4] d4 o3'=o5
+        4: o4 r4 d4 o4'=o5
```

Comment:
should we just drop all the frames in the range of [len-n+1:], since under n-step scheme, they don't technically have valid next_obs?
I am actually not sure about the standard algorithm at all. just feel like that may make things cleaner?

Reply:
See comment above. It wouldn't be good to drop them. The reasoning here is: Better to have a few n-step=1|2 SARS' tuples in the batch along with all the n-step=3 SARS' tuples, than to drop these shorter ones. Best would actually be to mix IMO.

Reply:
oh ok. completely trust you here :)